packages/backend/src/controllers/tasks/tasks.controller.ts

```diff
@@ -0,0 +1,295 @@
+    private readonly taskStoreService: TaskStoreService,
+    private readonly taskUtil: TaskUtil,
+  ) {}
+
+  @Post()
+  @ApiOperation({ summary: 'create a new task' })
+  @ApiCreatedResponse({
+    description: 'task created successfully',
+  })
+  @ApiBadRequestResponse({ description: 'Invalid request' })
+  async createTask(
+    @Body() createTaskDto: CreateTaskDto,
+    @User() user: DecodedIdToken,
+  ) {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+    if (!userDoc.house) {
+      throw new HttpException('user is not in a house', HttpStatus.BAD_REQUEST);
+    }
+
+    const house = await this.houseStoreService.findOne(userDoc.house);
+    if (house) {
+      if (createTaskDto.pool.length < 1) {
+        throw new HttpException('pool is empty', HttpStatus.BAD_REQUEST);
+      }
+
+      createTaskDto.assigned = this.taskUtil.selectRandomUser(
+        createTaskDto.pool,
+      );
+      createTaskDto.house = house._id;
+
+      await this.taskStoreService.create(createTaskDto);
+    }
+  }
+
+  @Get()
+  @ApiOperation({ summary: 'get the current tasks from a house' })
+  @ApiOkResponse({
+    description: 'tasks retrieved successfully',
+    type: HouseTasksResponseDto,
+  })
+  @ApiBadRequestResponse({ description: 'user is not in a house' })
+  async getTasksForHouse(
+    @User() user: DecodedIdToken,
+  ): Promise<HouseTasksResponseDto | null> {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+    if (!userDoc.house) {
+      throw new HttpException('user is not in a house', HttpStatus.BAD_REQUEST);
+    }
+
+    const house = await this.houseStoreService.findOne(userDoc.house);
+    if (house) {
+      const tasks = await this.taskStoreService.findAll();
+
+      const tasksForHouse = tasks.filter((task) =>
+        task.house.equals(house._id),
+      );
+
+      const tasksDue = this.taskUtil.checkRecurrence(tasksForHouse);
+      const taskPromises = tasksDue.map((task) =>
+        this.taskStoreService.update(task.id, task.updatedTask),
+      );
+      await Promise.all(taskPromises);
+
+      const updatedTasks = await this.taskStoreService.findAll();
+      const updatedTasksForHouse = updatedTasks.filter((task) =>
+        task.house.equals(house._id),
+      );
+
+      const updatedTasksDto: TaskResponseDto[] = updatedTasksForHouse.map(
+        (task) => {
+          const {
+            name,
+            description,
+            lastCompleted,
+            dueDate,
+            interval,
+            assigned,
+            pool,
+            house,
+          } = task;
+
+          return {
+            name,
+            description,
+            lastCompleted,
+            dueDate,
+            interval,
+            assigned,
+            pool,
+            house,
+            isComplete: task.lastCompleted != null,
+          };
+        },
+      );
+      return {
+        tasks: updatedTasksDto,
+      };
+    }
+  }
```

Comment:
can you destructure task instead directly? Sadly it looks like you can't destructure and mind at the same time like you can in other langs (binding@{prop1, prop2})

Reply:
oh there is a mistake i didnt, want to include the lastCompleted field

Reply:
actaully no i lie